src/cli/facade/publish.ts

```diff
@@ -152,7 +154,10 @@ const publish = ({
       ) => {
         for (const registryUrl of registryLocations) {
           const registryNormalised = registryUrl.replace(/\/$/, '');
-          const componentRoute = `${registryNormalised}/${component.name}/${component.version}`;
+          let componentRoute = `${registryNormalised}/${component.name}/${component.version}`;
+          if (dryRun) {
+            componentRoute += '?checkOnly';
```

Comment:
should we call it `dryRun`, or `dry` as well?

Reply:
The thing is to me, dryRun is context dependant. When you are on the CLI you just dry running, but when you call an API there is not really a concept of a dry run, and something like "only check" makes more sense, thats why the two terms for the two contexts: CLI / API.

Reply:
hmm... onlyCheck feels quite clunky to be honest, where, in fact, we're asking the server to do a dry-run as well, setting expectation of no side-effects